apple/Handlers/RNLongPressHandler.m

```diff
@@ -94,6 +107,83 @@ - (void)touchesCancelled:(NSSet<RNGHUITouch *> *)touches withEvent:(UIEvent *)ev
   [self reset];
 }
 
+#else
+- (void)mouseDown:(NSEvent *)event
+{
+  self.state = NSGestureRecognizerStateBegan;
+  startTime = CACurrentMediaTime();
+
+  [_gestureHandler.pointerTracker touchesBegan:[NSSet setWithObject:event] withEvent:event];
+
+  _initPosition = [self locationInView:self.view];
+
+  __weak typeof(self) weakSelf = self;
+
+  block = dispatch_block_create(0, ^{
+    __strong typeof(self) strongSelf = weakSelf;
+
+    if (strongSelf) {
+      strongSelf.state = NSGestureRecognizerStateChanged;
+      strongSelf.state = NSGestureRecognizerStateEnded;
+    }
+  });
+
+  dispatch_after(
+      dispatch_time(DISPATCH_TIME_NOW, (int64_t)(self.minimumPressDuration * NSEC_PER_SEC)),
+      dispatch_get_main_queue(),
+      block);
+}
+
+- (void)mouseDragged:(NSEvent *)event
+{
+  [_gestureHandler.pointerTracker touchesMoved:[NSSet setWithObject:event] withEvent:event];
+
+  if (block == nil) {
+    return;
+  }
+
+  if ([self shouldCancelGesture]) {
+    dispatch_block_cancel(block);
+    block = nil;
+
+    [self failOrCancelGesture];
+  }
+}
+
+- (void)mouseUp:(NSEvent *)event
+{
+  [_gestureHandler.pointerTracker touchesEnded:[NSSet setWithObject:event] withEvent:event];
+
+  if (block) {
+    dispatch_block_cancel(block);
+    block = nil;
+  }
+
+  [self failOrCancelGesture];
+}
+
+- (void)failOrCancelGesture
+{
+  if (self.state == NSGestureRecognizerStateChanged) {
+    self.state = NSGestureRecognizerStateCancelled;
+  } else {
+    self.state = NSGestureRecognizerStateFailed;
```

Comment:
What happens when you change the state from `ended` to `failed`? I assume that's what's happening when letting go of the mouse button after long press finishes successfully.

Reply:
I'm not exactly sure what happens, but I've noticed that `success` in `onFinalize` is set to `false` even if gesture has ended correctly. I've fixed that and turns out that it was because of this `if`. ([bab5b11](https://github.com/software-mansion/react-native-gesture-handler/pull/3014/commits/bab5b112b8e1161bc0ec81e5a4e1b8305585c8b6))